scrape/scrape.go

```diff
@@ -1553,7 +1553,15 @@ type appendErrors struct {
 }
 
 func (sl *scrapeLoop) append(app storage.Appender, b []byte, contentType string, ts time.Time) (total, added, seriesAdded int, err error) {
-	p, err := textparse.New(b, contentType, sl.fallbackScrapeProtocol, sl.alwaysScrapeClassicHist, sl.enableCTZeroIngestion, sl.symbolTable)
+	var p textparse.Parser
+
+	if len(b) == 0 {
+		// An empty scrape (such as from a timeout) should not cause errors to be logged or the append to fail.
+		// We treat it as if we received "text/plain" and allow the appender to continue to trigger stale markers.
+		p, err = textparse.New(b, "text/plain", "", sl.alwaysScrapeClassicHist, sl.enableCTZeroIngestion, sl.symbolTable)
```

Comment:
When you need twice as much comment as code, it's a warning flag.

Alternative suggestion: extract the `forEachStale` loop at the end of `append()` out to a new function, call it here, then return.
Then you could lose the `b > 0` check and comment from 1627-1629.

Reply:
Thanks, have reworked as suggested.